File: forum-backend/src/main/java/com/projects/pes/forumbackend/pojo/Faculty.java
package com.projects.pes.forumbackend.pojo;

import com.projects.pes.forumbackend.entities.ForumEntity;
import lombok.Data;

import java.util.List;
import java.util.Set;
import java.util.UUID;

@Data
public class Faculty extends User {
        public Faculty() {
            super();
            setRole(UserRole.FACULTY);
        }
        public Faculty (
                UUID id,
                String username,
                String email,
                String name,
                String password,
                String imgUrl,
                String contact,
                Set<Forum> forums,
                Section section,
                String jobTitle,
                String department,
                List<String>domains,
                List<String> publications
        ) {
            super(id,username,email,name,password,imgUrl,section,contact,forums,UserRole.FACULTY);
            this.jobTitle = jobTitle;
            this.department = department;
            this.domains = domains;
            this.publications = publications;
        }
        private String jobTitle;
        private String department;
        private List<String> domains;
        private List<String> publications;
    }

File: student-forums-frontend/src/app/faculty.ts
import { User } from "./user";
import { Forum } from "./forum";
export class Faculty extends User {
    jobTitle:string|null;
    department:string;
    domains:Array<string>;
    publications!:Array<string>;
    constructor(
        id:string|null,
        username:string,
        password:string,
        email:string,
        name:string,
        imgUrl:string|null,
        jobTitle:string|null,
        department:string,
        domains:Array<string>,
        publications:Array<string>,
        contact:string|null,
    ) {
        super(id!,username, password, email, name);
        this.jobTitle = jobTitle;
        this.department = department;
        this.domains = domains;
        this.publications = publications;
        this.imgUrl = imgUrl;
        this.contact = contact;
    
    }

    
}
